certbot/cert_manager.py

```diff
@@ -95,6 +98,14 @@ def delete(config):
 # Public Helpers
 ###################
 
+def full_archive_dir_from_renewal_conf(cli_config):
+    """Normally the sane thing to do is to use storage.full_archive_path,
+    but this function exists to help perform consistency checks preventing
+    users who set multiple different lineages to use the same archive dir."""
+    renewal_conf = storage.renewal_file_for_certname(cli_config, cli_config.certname)
```

Comment:
It probably still makes sense to use `_full_archive_path`. `archive_dir` was added to renewal files at some point. So, while falling back to the default value means there definitely won't be an overlap, it's better to just have that and know that there isn't one than to erroneously error out here.